### ISSUE TITLE
android9.0 Chrash

### ISSUE THREAD

**Issue by RmondJone** [iou90/react-native-autoheight-webview#117]
```java
  @ReactProp(name = "androidHardwareAccelerationDisabled")
  public void setHardwareAccelerationDisabled(WebView view, boolean disabled) {
    if (disabled) {
      view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
    } else {
      view.setLayerType(View.LAYER_TYPE_NONE, null);
    }
  }
```
The tracking code is because the hardware acceleration is not enabled and the setting on the JS side is invalid.

**Comment by iou90** [comment]
Does this issue exists only with react-native-webview?

**Comment by RmondJone** [comment]
Yes,And I solved this
```java
  "scripts": {
    "rn-start": "node node_modules/react-native/local-cli/cli.js start",
    "start": "node scripts/custom-rn-theme && npm run rn-start",
    "scheme": "node scripts/custom-rn-theme",
    "build-android-executor": "cd android && gradlew installRelease",
    "da": "cd android && gradlew installDebug",
    "bundle-android": "node scripts/custom-rn-theme && npm run build-android-executor",
    "build-ios-executor": "react-native bundle --entry-file index.ios.js --bundle-output ./ios/bundle/index.ios.jsbundle --platform ios --assets-dest ./ios/bundle --dev false",
    "bundle-ios": "node scripts/custom-rn-theme && npm run build-ios-executor",
    "test": "jest",
    "fix-rn": "node scripts/fix-rn",
    "postinstall": "npm run fix-rn && node node_modules/react-native-autotrack-growingio/hook.js -run"
  },
```
And fix-rn
```java
const path = require('path');
const fs = require('fs');

const webviewPath = require.resolve('react-native-webview')
const webviewRootPath = path.join(webviewPath, '../')

const webviewVMPath = path.join(__dirname, 'files', 'RNCWebViewManager.java')
const webviewVMToPath = path.join(webviewRootPath, 'android/src/main/java/com/reactnativecommunity/webview', 'RNCWebViewManager.java')

moveFile(webviewVMPath, webviewVMToPath)

function moveFile(sourcePath, destPath) {
  if (fs.existsSync(destPath)) {
    fs.unlinkSync(destPath);
  }
  const readStream = fs.createReadStream(sourcePath);
  const writeStream = fs.createWriteStream(destPath);
  readStream.pipe(writeStream);
  console.log(`copy---${sourcePath} to \n --- ${destPath} `)
}
```
![image](https://user-images.githubusercontent.com/10555850/60149315-d9094580-9806-11e9-9f6b-1880dcc3cb83.png)

**Comment by vkozinec** [comment]
Hello, we are experiencing same issue on Android 9, when using this library, no matter if `<AutoHeightWebView/>`is used in modal or just on plain screen. 

However, when using react-native-webview itself on other screens, plain `<WebView>` works normally.

So I wasn't sure if problem was with webview itself. 

Just wondering, if top answer is problem solution, will it be included as fix in library or we have to do it manually. 

Thank you.

**Comment by RmondJone** [comment]
> Hello, we are experiencing same issue on Android 9, when using this library, no matter if `<AutoHeightWebView/>`is used in modal or just on plain screen.
> 
> However, when using react-native-webview itself on other screens, plain `<WebView>` works normally.
> 
> So I wasn't sure if problem was with webview itself.
> 
> Just wondering, if top answer is problem solution, will it be included as fix in library or we have to do it manually.
> 
> Thank you.

Look Up

**Comment by iou90** [comment]
@RmondJone 
Your codes within RNCWebViewManager.java are the same as the codes of react-native-webview since this pr https://github.com/react-native-community/react-native-webview/commit/c8977ec7cefebb8e0d6ca6bb6216428353d272c5#diff-186a7fa0ea29d9ede1ed98c48d63340f.

@vkozinec 
I can not reproduction this issue with the demo of this repo on emulator (api28, Android Pie / 9.0), could you provide more information about your case?

**Comment by vkozinec** [comment]
So we have:

`react-native-webview` v**5.8.1**
`compileSdkVersion = 28`
`"react-native": "^0.59.8"`

I have updated `react-native-webview` to **6.3.0** and added to gradle.properties 
```
android.useAndroidX=true
android.enableJetifier=true
```

and after that android build is failing. Looks like it has something to do with `react-native` version, since AndroidX is added in `0.60.x` and we are still on `0.59.8`

**Comment by iou90** [comment]
@vkozinec 
The pr react-native-community/react-native-webview@c8977ec#diff-186a7fa0ea29d9ede1ed98c48d63340f included in react-native-webview@5.8.2: https://github.com/react-native-community/react-native-webview/releases/tag/v5.8.2.
And the version of react-native-webview in demo is 5.12.0.

**Comment by otusweb** [comment]
I"m curious what the right solution is here. I'm running into the same issue (crash on android 9) in certain case when using autoheight webview.
What is the trail to follow to fix this on ym build :-)

**Comment by iou90** [comment]
@otusweb 
Some details about your case?

**Comment by vkozinec** [comment]
@otusweb we have workaround for problem for now.

Disabling animations in WIX navigation when navigating on next screen fixes problem for now. We are still not on the latest version of RN, so when we upgrade, we will test again.

**Comment by otusweb** [comment]
"react-native": "0.59.10",
"react-native-autoheight-webview": "1.1.4",
"react-native-webview": "5.12.1"
Android 9


note that there seem to be a timing issue. If I step through the code with various breakpoints, it does not crash. If i just run it in the debugger with no breakpoint, it does not crash.

Html i'm trying to render:
```
<HTML><header><meta name="viewport" content="initial-scale=1.0">
<style> 
html * { 
  font-family: 'AppleSDGothicNeo-Regular';
}
p {
  margin: 0 0 4px;
}
body  {
  color: #3A4C58;
  font-weight: 400;
}
b, strong {
  font-weight: 700;
}

em {
  font-style: italic;
}

#rnahw-wrapper{ 
  padding-bottom: 20px;
}

blockquote {
  padding: 10px 20px;
  margin: 0 0 20px;
  font-size: 17.5px;
  border-left: 5px solid #eee;
}

blockquote ol:last-child, blockquote p:last-child, blockquote ul:last-child {
  margin-bottom: 0;
}

.h1, .h2, .h3, h1, h2, h3 {
  margin-top: 10px;
  margin-bottom: 10px;
}

.h1, .h2, .h3, .h4, .h5, .h6, h1, h2, h3, h4, h5, h6 {
  font-family: inherit;
  font-weight: 500;
  line-height: 1.1;
  color: inherit;
}
</style></header><body><p>bla bla</p></body></HTML>
```

my wrapper around autoheight webview
```
// @flow
import React from "react";
import { Platform } from "react-native";
import AutoHeightWebView from "react-native-autoheight-webview";
import { Navigation } from "react-native-navigation";

//project import

type Props = {
  html: string,
  style: any,
  customStyle?: ?string
};
type State = {};
class WebView extends React.Component<Props, State> {
  //TODO: $FlowFixMe
  webViewRef: React.RefObject<AutoHeightWebView>;
  constructor(props: Props) {
    super(props);

    this.webViewRef = React.createRef();
    this.state = {};
  }

  handleShouldStartLoadWithRequest = (event: any) => {
    //this is only called on iOS
    let url = event.url;
    return this.shouldKeepLoading(url);
  };

  handleLoadStart = (event: any) => {
    //handle android side...
    if (Platform.OS == "android") {
      let url = event.nativeEvent.url;
      return this.shouldKeepLoading(url);
    } else {
      return true;
    }
  };

  shouldKeepLoading = (url: string) => {
    if (url.startsWith("http")) {
      const webview = this.webViewRef.current;
      webview.stopLoading();

      //the line below is only for android, but it has no side effect on iOS
      webview.webView.current.goBack();

      Navigation.showModal({
        stack: {
          children: [
            {
              component: {
                name: "WazzaScreen.WebBrowser",
                passProps: {
                  url: url
                },
                options: {
                  topBar: {
                    backButton: {
                      title: ""
                    }
                  }
                }
              }
            }
          ]
        }
      });

      return false;
    } else {
      return true;
    }
  };

  render = () => {
    return (
      <AutoHeightWebView
        ref={this.webViewRef}
        source={{
          html: this.props.html
        }}
        scalesPageToFit={Platform.OS === "ios" ? false : true}
        scrollEnabled={false}
        style={this.props.style}
        onShouldStartLoadWithRequest={this.handleShouldStartLoadWithRequest}
        showsVerticalScrollIndicator={false}
        onLoadStart={this.handleLoadStart}
        customStyle={this.props.customStyle}
      />
    );
  };
}

export default WebView;
```

JSX
```
 <View style={SharedStyleSheet.screenContainer}>
        <AnonymousConnection />
        <ScrollView
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator
          style={{
            alignSelf: "stretch"
          }}
          contentContainerStyle={[
            {
              justifyContent: "flex-start",
              alignItems: "stretch"
            }
          ]}
        >
          {imageURLs.length != 0 ? (
            <View>
              <Image
                source={{ uri: message.images[0].url }}
                style={[{ width: "100%" }]}
                width={Dimensions.get("window").width}
                height={275}
                resizeMode="cover"
                onError={this.handleImageLoadError}
              />
              <BackButton
                style={{
                  backgroundColor: "rgba(0, 0, 0, 0.4)",
                  borderRadius: 20,
                  paddingLeft: 10,
                  position: "absolute",
                  top: 10,
                  left: 20
                }}
                lightIcon
                action={this.onClose}
              />
            </View>
          ) : (
            <BackButton
              style={{
                marginTop: 10,
                marginLeft: 20
              }}
              action={this.onClose}
            />
          )}

          <View style={{ padding: 20, paddingTop: 10, alignItems: "stretch" }}>
[...]
            <WebView html={messageBody} style={styles.webViewStyle} />
          </View>
        </ScrollView>
      </View>
```

**Comment by iou90** [comment]
@otusweb  Does the problem still exists without react-native-navigation?

**Comment by otusweb** [comment]
@iou90 I don't know as it is the navigation system integrated in the app. It clearly seems to happen with the screen transition. Unfortunately, I don't have a good way to test with another navigation system

**Comment by iou90** [comment]
Test without 
```
Navigation.showModal({
        stack: {
          children: [
            {
              component: {
                name: "WazzaScreen.WebBrowser",
                passProps: {
                  url: url
                },
                options: {
                  topBar: {
                    backButton: {
                      title: ""
                    }
                  }
                }
              }
            }
          ]
        }
      });
```
@otusweb

**Comment by otusweb** [comment]
@iou90 the crash happen even without that code, but this show modal is never called in the crash sequence, the web browser that crashes is shown with:
```
Navigation.push(this.props.componentId, {
      component: {
        name: "WazzaScreen.messageDetails",
        passProps: {
          message: message
        },
        options: {
          topBar: {
            backButton: {
              title: ""
            }
          }
        }
      }
    });
```

I'm not sure how to show that screen without calling navigation.push

The showModal above only happens if the user clicks on  a link shown in the original browser that cashes

**Comment by iou90** [comment]
Please use react-native-webview with fixed height in your component and check whether the issue exists.

**Comment by vkozinec** [comment]
> Please use react-native-webview with fixed height in your component and check whether the issue exists.

When using AutoHeightWebView with fixed width from style, crash is NOT happening. 

For now, we have disabled animations in WIX navigation on screens where we are using AutoHeightWebView and it works.

Also when using AutoHeightWebView with `flex:1` and using whole screen, there is no crash.

**Comment by rajscola** [comment]
Enabling androidHardwareAccelerationDisabled worked for me. Thanks to this [bug report](https://github.com/react-native-community/react-native-webview/issues/575)

**Comment by iou90** [comment]
https://github.com/react-native-community/react-native-webview/issues/575#issuecomment-533922767 resolved with react-native-webview@7.1.1
